packages/core/src/components/Box/react/box.test.tsx

```diff
@@ -17,4 +17,13 @@ describe("Box component", () => {
       color: "rgb(255, 255, 255)",
     });
   });
+
+  it("should correctly override prop types by `as` prop", () => {
+    // Arrange
+    const ref = createRef<HTMLButtonElement>();
+    const onClick: MouseEventHandler<HTMLButtonElement> = () => {};
+
+    // Assert
+    <Box as="button" ref={ref} onClick={onClick} />;
+  });
```

Comment:
I'm not familiar with type test. Tell me if you know any better way!

Reply:
How about this one? https://github.com/kuma-ui/kuma-ui/pull/351/files#r1342149178